hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/RunCompactionActionExecutor.java

```diff
@@ -81,7 +82,7 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
         metadata.addWriteStat(stat.getPartitionPath(), stat);
       }
       metadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, config.getSchema());
-
+      metadata.setOperationType(WriteOperationType.COMPACT);
       compactionMetadata.setWriteStatuses(statuses);
       compactionMetadata.setCommitted(false);
```

Comment:
Hmm, a very long time, i thought this was a design purpose, the compaction uses the commit action time as COW on the timeline, we can also see this as a bug.